homeassistant/components/slide_local/config_flow.py

```diff
@@ -0,0 +1,180 @@
+        slide = SlideLocalApi()
+
+        # first test, if API version 2 is working
+        await slide.slide_add(
+            user_input[CONF_HOST],
+            user_input.get(CONF_PASSWORD, ""),
+            2,
+        )
+
+        try:
+            result = await slide.slide_info(user_input[CONF_HOST])
+        except (ClientConnectionError, ClientTimeoutError):
+            return {"base": "cannot_connect"}
+        except (AuthenticationFailed, DigestAuthCalcError):
+            return {"base": "invalid_auth"}
+        except Exception:  # noqa: BLE001
+            _LOGGER.exception("Exception occurred during connection test")
+            return {"base": "unknown"}
+
+        if result is not None:
+            self._api_version = 2
+            self._mac = result["mac"]
+            return {}
+
+        # API version 2 is not working, try API version 1 instead
+        await slide.slide_del(user_input[CONF_HOST])
+        await slide.slide_add(
+            user_input[CONF_HOST],
+            user_input.get(CONF_PASSWORD, ""),
+            1,
+        )
+
+        try:
+            result = await slide.slide_info(user_input[CONF_HOST])
+        except (ClientConnectionError, ClientTimeoutError):
+            return {"base": "cannot_connect"}
+        except (AuthenticationFailed, DigestAuthCalcError):
+            return {"base": "invalid_auth"}
+        except Exception:  # noqa: BLE001
+            _LOGGER.exception("Exception occurred during connection test")
+            return {"base": "unknown"}
+
+        if result is None:
+            # API version 1 isn't working either
+            return {"base": "unknown"}
+
+        self._api_version = 1
+        self._mac = result["mac"]
+
+        return {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the user step."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+
+            if not (errors := await self.async_test_connection(user_input)):
+                await self.async_set_unique_id(self._mac)
+                user_input |= {
+                    CONF_MAC: format_mac(self._mac),
+                    CONF_API_VERSION: self._api_version,
+                }
+
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST],
+                    data=user_input,
+                    options={CONF_INVERT_POSITION: False},
+                )
+
+        if user_input is not None and user_input.get(CONF_HOST) is not None:
+            self._host = user_input[CONF_HOST]
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self._host): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
```

Comment:
can we use `self.add_suggested_values_to_schema` here instead of setting a `default`?